desc/basis.py

```diff
@@ -1595,134 +1661,440 @@ def fourier(theta, m, NFP=1, dt=0):
     return m_abs**dt * jnp.sin(m_abs * theta + shift)
 
 
-@jit
-@jnp.vectorize
-def _binom(n, k):
-    """Binomial coefficient.
-
-    Implementation is only correct for positive integer n,k and n>=k
+def zernike_norm(l, m):
+    """Norm of a Zernike polynomial with l, m indexing.
 
     Parameters
     ----------
-    n : int, array-like
-        number of things to choose from
-    k : int, array-like
-        number of things chosen
+    l,m : int
+        radial and azimuthal mode numbers.
 
     Returns
     -------
-    val : int, float, array-like
-        number of possible combinations
+    norm : float
+        the integral (Z^m_l)^2 r dr dt, r=[0,1], t=[0,2*pi]
+
     """
-    # adapted from scipy:
-    # https://github.com/scipy/scipy/blob/701ffcc8a6f04509d115aac5e5681c538b5265a2/
-    # scipy/special/orthogonal_eval.pxd#L68
+    return np.sqrt((2 * (l + 1)) / (np.pi * (1 + int(m == 0))))
```

Comment:
test for this function?

Reply:
I don't even remember coding this :D

Reply:
This is (very) old code, it just got moved around. Iirc there is already a test for it

Reply:
Oh yeah it is 3 years old. Definitely not me!